jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -122,11 +129,13 @@ public static JaegerSpanContext contextFromString(String value)
       oibe: because java doesn't like to convert large hex strings to longs
       we should write this manually instead of using BigInteger.
     */
-    return new JaegerSpanContext(
+    return factory.createSpanContext(
         new BigInteger(parts[0], 16).longValue(),
         new BigInteger(parts[1], 16).longValue(),
         new BigInteger(parts[2], 16).longValue(),
-        new BigInteger(parts[3], 16).byteValue());
+        new BigInteger(parts[3], 16).byteValue(),
+        Collections.<String, String>emptyMap(),
+        null);
```

Comment:
what is null?  should it be `factory`?

Reply:
This is actually a reference to `debugId`. But this should definitely take a `TracingFactory` argument as well. Will fix.

Reply:
Never mind. The factory passes `this` internally to the `JaegerSpanContext` constructor.